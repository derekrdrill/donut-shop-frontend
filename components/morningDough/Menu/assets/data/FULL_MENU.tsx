import avocadoToast from '../images/bread/avocadoToast.png';
import blueberryMuffin from '../images/bread/blueberryMuffin.png';
import chocolateMuffin from '../images/bread/chocolateMuffin.png';
import croissant from '../images/bread/croissant.png';
import everythingBagel from '../images/bread/everythingBagel.png';
import plainBagel from '../images/bread/plainBagel.png';
import poppyBagel from '../images/bread/poppyBagel.png';
import raisinBagel from '../images/bread/raisinBagel.png';
import cocaCola from '../images/coffeeAndTea/cocaCola.png';
import hotCoffee from '../images/coffeeAndTea/hotCoffee.png';
import hotLatte from '../images/coffeeAndTea/hotLatte.png';
import hotTea from '../images/coffeeAndTea/hotTea.png';
import icedCoffee from '../images/coffeeAndTea/icedCoffee.png';
import icedTea from '../images/coffeeAndTea/icedTea.png';
import sprite from '../images/coffeeAndTea/sprite.png';
import waterBottle from '../images/coffeeAndTea/waterBottle.png';
import bostonCreamDonut from '../images/donuts/bostonCreamDonut.png';
import chocolateDonut from '../images/donuts/chocolateDonut.png';
import chocolateFrostingDonut from '../images/donuts/chocolateFrostingDonut.png';
import chocolateFrostingSprinklesDonut from '../images/donuts/chocolateFrostingSprinklesDonut.png';
import glazedDonut from '../images/donuts/glazedDonut.png';
import jellyFilledDonut from '../images/donuts/jellyFilledDonut.png';
import powderedDonut from '../images/donuts/powderedDonut.png';
import sprinkleDonut from '../images/donuts/sprinkleDonut.png';

export interface FullMenuItem {
  key: string;
  name: string;
  image: string;
  category: string;
  subCategory: string;
  soda?: boolean;
  bottled?: boolean;
}

const FULL_MENU: FullMenuItem[] = [
  {
    key: 'chocolateDonut',
    name: 'Chocolate Donut',
    image: chocolateDonut,
    category: 'donuts',
    subCategory: 'donuts',
  },
  {
    key: 'glazedDonut',
    name: 'Glazed Donut',
    image: glazedDonut,
    category: 'donuts',
    subCategory: 'donuts',
  },
  {
    key: 'powderedDonut',
    name: 'Powdered Donut',
    image: powderedDonut,
    category: 'donuts',
    subCategory: 'donuts',
  },
  {
    key: 'sprinkleDonut',
    name: 'Strawberry sprinkles donut',
    image: sprinkleDonut,
    category: 'donuts',
    subCategory: 'donuts',
  },
  {
    key: 'chocolateFrostingDonut',
    name: 'Chocolate frosting donut',
    image: chocolateFrostingDonut,
    category: 'donuts',
    subCategory: 'donuts',
  },
  {
    key: 'chocolateFrostingSprinklesDonut',
    name: 'Chocolate w/ sprinkles donut',
    image: chocolateFrostingSprinklesDonut,
    category: 'donuts',
    subCategory: 'donuts',
  },
  {
    key: 'jellyFilledDonut',
    name: 'Jelly Filled Donut',
    image: jellyFilledDonut,
    category: 'donuts',
    subCategory: 'donuts',
  },
  {
    key: 'bostonCreamDonut',
    name: 'Boston Cream Donut',
    image: bostonCreamDonut,
    category: 'donuts',
    subCategory: 'donuts',
  },
  {
    key: 'plainBagel',
    name: 'Plain Bagel',
    image: plainBagel,
    category: 'bread',
    subCategory: 'bagels',
  },
  {
    key: 'poppyBagel',
    name: 'Poppy Bagel',
    image: poppyBagel,
    category: 'bread',
    subCategory: 'bagels',
  },
  {
    key: 'everythingBagel',
    name: 'Everything Bagel',
    image: everythingBagel,
    category: 'bread',
    subCategory: 'bagels',
  },
  {
    key: 'raisinBagel',
    name: 'Raisin Bagel',
    image: raisinBagel,
    category: 'bread',
    subCategory: 'bagels',
  },
  {
    key: 'blueberryMuffin',
    name: 'Blueberry Muffin',
    image: blueberryMuffin,
    category: 'bread',
    subCategory: 'muffins',
  },
  {
    key: 'chocolateMuffin',
    name: 'Chocolate Muffin',
    image: chocolateMuffin,
    category: 'bread',
    subCategory: 'muffins',
  },
  {
    key: 'croissant',
    name: 'Croissant',
    image: croissant,
    category: 'bread',
    subCategory: 'croissants',
  },
  {
    key: 'avocadoToast',
    name: 'Avocado Toast',
    image: avocadoToast,
    category: 'bread',
    subCategory: 'toast',
  },
  {
    key: 'hotCoffee',
    name: 'Hot Coffee',
    image: hotCoffee,
    category: 'drink',
    subCategory: 'hotDrinks',
  },
  {
    key: 'icedCoffee',
    name: 'Iced Coffee',
    image: icedCoffee,
    category: 'drink',
    subCategory: 'coldDrinks',
  },
  {
    key: 'hotLatte',
    name: 'Hot Latte',
    image: hotLatte,
    category: 'drink',
    subCategory: 'hotDrinks',
  },
  { key: 'hotTea', name: 'Hot Tea', image: hotTea, category: 'drink', subCategory: 'hotDrinks' },
  {
    key: 'icedTea',
    name: 'Iced Tea',
    image: icedTea,
    category: 'drink',
    subCategory: 'coldDrinks',
    soda: true,
  },
  {
    key: 'cocaCola',
    name: 'Coca-Cola',
    image: cocaCola,
    category: 'drink',
    subCategory: 'coldDrinks',
    soda: true,
  },
  {
    key: 'sprite',
    name: 'Sprite',
    image: sprite,
    category: 'drink',
    subCategory: 'coldDrinks',
    soda: true,
  },
  {
    key: 'waterBottle',
    name: 'Water Bottle',
    image: waterBottle,
    category: 'drink',
    subCategory: 'coldDrinks',
    bottled: true,
  },
];

export default FULL_MENU;
