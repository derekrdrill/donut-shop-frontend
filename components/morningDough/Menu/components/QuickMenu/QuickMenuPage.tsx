import * as React from 'react';
import { Grid, Typography } from '@mui/material';

import QuickMenuItems from './components/QuickMenuItems';

import { QuickMenuTabContainer } from './style';

import { FullMenuItem } from '../../assets/data/FULL_MENU';

import FULL_MENU from '../../assets/data/FULL_MENU';

export const getQuickMenuData = (quickMenuView: string) =>
  quickMenuView === 'coffee'
    ? FULL_MENU.filter(menuItem => menuItem.category === 'drink')
    : quickMenuView === 'donuts'
    ? FULL_MENU.filter(menuItem => menuItem.category === 'donuts')
    : FULL_MENU.filter(menuItem => menuItem.category === 'bread');

interface QuickMenuPageProps {
  setIsQuickMenuOpen: React.Dispatch<React.SetStateAction<boolean>>;
}

const QuickMenuPage = ({ setIsQuickMenuOpen }: QuickMenuPageProps) => {
  const [quickMenuView, setQuickMenuView] = React.useState<string>('donuts');
  const [quickMenuData, setQuickMenuData] = React.useState<FullMenuItem[]>(
    FULL_MENU.filter(menuItem => menuItem.category === 'drink'),
  );

  React.useEffect(() => {
    setQuickMenuData(getQuickMenuData(quickMenuView));
  }, [quickMenuView]);

  return (
    <Grid container>
      <Grid item xs={12}>
        <Grid container>
          <Grid item xs={4}>
            <QuickMenuTabContainer
              container
              justifyContent='center'
              onClick={
                /* istanbul ignore next */
                () => setQuickMenuView('coffee')
              }
              $isMenuTabActive={quickMenuView === 'coffee'}
            >
              <Typography variant='subtitle1'>Coffee, Tea & More</Typography>
            </QuickMenuTabContainer>
          </Grid>
          <Grid item xs={4}>
            <QuickMenuTabContainer
              container
              justifyContent='center'
              onClick={
                /* istanbul ignore next */
                () => setQuickMenuView('donuts')
              }
              $isMenuTabActive={quickMenuView === 'donuts'}
            >
              <Typography variant='subtitle1'>Doughnuts</Typography>
            </QuickMenuTabContainer>
          </Grid>
          <Grid item xs={4}>
            <QuickMenuTabContainer
              container
              justifyContent='center'
              onClick={
                /* istanbul ignore next */
                () => setQuickMenuView('bread')
              }
              $isMenuTabActive={quickMenuView === 'bread'}
            >
              <Typography variant='subtitle1'>Bagels, Muffins & More</Typography>
            </QuickMenuTabContainer>
          </Grid>
        </Grid>
      </Grid>
      <Grid item xs={12}>
        <QuickMenuItems quickMenuData={quickMenuData} setIsQuickMenuOpen={setIsQuickMenuOpen} />
      </Grid>
    </Grid>
  );
};

export default QuickMenuPage;
