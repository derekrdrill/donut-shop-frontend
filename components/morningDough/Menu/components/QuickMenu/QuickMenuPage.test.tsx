jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => jest.fn(),
}));

import '@testing-library/jest-dom';
import * as React from 'react';
import * as renderer from 'react-test-renderer';

import QuickMenuPage, { getQuickMenuData } from './QuickMenuPage';

import FULL_MENU from '../../assets/data/FULL_MENU';

describe('QuickMenuPage unit tests', () => {
  it('renders as expected', () => {
    const result = renderer.create(<QuickMenuPage setIsQuickMenuOpen={jest.fn()} />).toJSON();
    expect(result).toMatchSnapshot();
  });

  it('renders getQuickMenuData correctly', () => {
    expect(getQuickMenuData('coffee')).toEqual(
      FULL_MENU.filter(menuItem => menuItem.category === 'drink'),
    );
    expect(getQuickMenuData('donuts')).toEqual(
      FULL_MENU.filter(menuItem => menuItem.category === 'donuts'),
    );
    expect(getQuickMenuData('bread')).toEqual(
      FULL_MENU.filter(menuItem => menuItem.category === 'bread'),
    );
  });
});
