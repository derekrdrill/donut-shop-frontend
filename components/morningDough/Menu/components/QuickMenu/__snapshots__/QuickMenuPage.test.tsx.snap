// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuickMenuPage unit tests renders as expected 1`] = `
<div
  className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
      >
        <div
          className="MuiGrid-root MuiGrid-container sc-gueYoa dshuxz css-1lym95h-MuiGrid-root"
          onClick={[Function]}
        >
          <h6
            className="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
          >
            Coffee, Tea & More
          </h6>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
      >
        <div
          className="MuiGrid-root MuiGrid-container sc-gueYoa hJcBfV css-1lym95h-MuiGrid-root"
          onClick={[Function]}
        >
          <h6
            className="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
          >
            Doughnuts
          </h6>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
      >
        <div
          className="MuiGrid-root MuiGrid-container sc-gueYoa dshuxz css-1lym95h-MuiGrid-root"
          onClick={[Function]}
        >
          <h6
            className="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
          >
            Bagels, Muffins & More
          </h6>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container sc-hLseeU exfbdM css-11lq3yg-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 sc-dmqHEX hfdwKv css-1o7apob-MuiGrid-root"
        onClick={[Function]}
      >
        <div
          className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
          >
            <div
              className="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
            >
              <img
                className="sc-beqWaB ETGuZ"
                src={
                  {
                    "process": [Function],
                  }
                }
              />
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
          >
            <div
              className="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
            >
              <h6
                className="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
              >
                Hot Coffee
              </h6>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 sc-dmqHEX hfdwKv css-1o7apob-MuiGrid-root"
        onClick={[Function]}
      >
        <div
          className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
          >
            <div
              className="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
            >
              <img
                className="sc-beqWaB ETGuZ"
                src={
                  {
                    "process": [Function],
                  }
                }
              />
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
          >
            <div
              className="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
            >
              <h6
                className="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
              >
                Iced Coffee
              </h6>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 sc-dmqHEX hfdwKv css-1o7apob-MuiGrid-root"
        onClick={[Function]}
      >
        <div
          className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
          >
            <div
              className="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
            >
              <img
                className="sc-beqWaB ETGuZ"
                src={
                  {
                    "process": [Function],
                  }
                }
              />
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
          >
            <div
              className="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
            >
              <h6
                className="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
              >
                Hot Latte
              </h6>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 sc-dmqHEX hfdwKv css-1o7apob-MuiGrid-root"
        onClick={[Function]}
      >
        <div
          className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
          >
            <div
              className="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
            >
              <img
                className="sc-beqWaB ETGuZ"
                src={
                  {
                    "process": [Function],
                  }
                }
              />
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
          >
            <div
              className="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
            >
              <h6
                className="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
              >
                Hot Tea
              </h6>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 sc-dmqHEX hfdwKv css-1o7apob-MuiGrid-root"
        onClick={[Function]}
      >
        <div
          className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
          >
            <div
              className="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
            >
              <img
                className="sc-beqWaB ETGuZ"
                src={
                  {
                    "process": [Function],
                  }
                }
              />
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
          >
            <div
              className="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
            >
              <h6
                className="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
              >
                Iced Tea
              </h6>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 sc-dmqHEX hfdwKv css-1o7apob-MuiGrid-root"
        onClick={[Function]}
      >
        <div
          className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
          >
            <div
              className="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
            >
              <img
                className="sc-beqWaB ETGuZ"
                src={
                  {
                    "process": [Function],
                  }
                }
              />
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
          >
            <div
              className="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
            >
              <h6
                className="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
              >
                Coca-Cola
              </h6>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 sc-dmqHEX hfdwKv css-1o7apob-MuiGrid-root"
        onClick={[Function]}
      >
        <div
          className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
          >
            <div
              className="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
            >
              <img
                className="sc-beqWaB ETGuZ"
                src={
                  {
                    "process": [Function],
                  }
                }
              />
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
          >
            <div
              className="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
            >
              <h6
                className="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
              >
                Sprite
              </h6>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 sc-dmqHEX hfdwKv css-1o7apob-MuiGrid-root"
        onClick={[Function]}
      >
        <div
          className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
          >
            <div
              className="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
            >
              <img
                className="sc-beqWaB ETGuZ"
                src={
                  {
                    "process": [Function],
                  }
                }
              />
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
          >
            <div
              className="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
            >
              <h6
                className="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
              >
                Water Bottle
              </h6>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
